<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Evgenii O. Tretiakov" />

<meta name="date" content="2024-07-25" />

<title>Methods</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cinquina et al., 2024</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="MTT.html">MTT assay analysis of EDA-treated astrocytes</a>
    </li>
    <li>
      <a href="cortex_visualisation.html">Exploratory analysis of CaCyBP and S100a6 expression in embryonic mice cortex dataset with focus on astrocytic lineage</a>
    </li>
  </ul>
</li>
<li>
  <a href="methods.html">Methods</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="https://github.com/Harkany-Lab/Cinquina_2024">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Methods</h1>
<h4 class="author">Evgenii O. Tretiakov</h4>
<h4 class="date">2024-07-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Cinquina_2024/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240320code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240320)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240320code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240320)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomHarkanyLabCinquina2024tree70b65cfae0097f7b6fd9ce7c9999aea44e439721targetblank70b65cfa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Harkany-Lab/Cinquina_2024/tree/70b65cfae0097f7b6fd9ce7c9999aea44e439721" target="_blank">70b65cf</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomHarkanyLabCinquina2024tree70b65cfae0097f7b6fd9ce7c9999aea44e439721targetblank70b65cfa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Harkany-Lab/Cinquina_2024/tree/70b65cfae0097f7b6fd9ce7c9999aea44e439721" target="_blank">70b65cf</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/SCP1290/
    Ignored:    data/azimuth_integrated.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/methods.Rmd</code>) and HTML
(<code>docs/methods.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Harkany-Lab/Cinquina_2024/blob/70b65cfae0097f7b6fd9ce7c9999aea44e439721/analysis/methods.Rmd" target="_blank">70b65cf</a>
</td>
<td>
Evgenii O. Tretiakov
</td>
<td>
2024-07-25
</td>
<td>
workflowr::wflow_publish(here::here("analysis/methods.Rmd"),
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Harkany-Lab/Cinquina_2024/blob/3b3abfe279aa73a1f49f13f95d09fe999139e8cd/analysis/methods.Rmd" target="_blank">3b3abfe</a>
</td>
<td>
EugOT
</td>
<td>
2024-07-25
</td>
<td>
add bibliography files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Harkany-Lab/Cinquina_2024/blob/78f7fe4d55c950ff8ca12d6bc847278a91f40f7b/analysis/methods.Rmd" target="_blank">78f7fe4</a>
</td>
<td>
Evgenii O. Tretiakov
</td>
<td>
2024-07-25
</td>
<td>
add description
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code># Presentation
library(&quot;glue&quot;)
library(&quot;knitr&quot;)

# JSON
library(&quot;jsonlite&quot;)

# Tidyverse
library(&quot;tidyverse&quot;)

library(Seurat)
library(dabestr)
library(here)
library(RColorBrewer)
library(scCustomize)
library(SeuratData)
library(SeuratWrappers)
library(Azimuth)
library(magrittr)
library(cowplot)
library(patchwork)</code></pre>
<pre class="r"><code>dir.create(here::here(&quot;output&quot;, DOCNAME), showWarnings = FALSE)

write_bib(c(&quot;base&quot;, &quot;Seurat&quot;, &quot;SeuratWrappers&quot;, &quot;SeuratData&quot;, &quot;sctransform&quot;,
            &quot;patchwork&quot;, &quot;scCustomize&quot;, &quot;cowplot&quot;, &quot;UpSetR&quot;, &quot;gridExtra&quot;, 
            &quot;tidyverse&quot;, &quot;dplyr&quot;, &quot;tidyr&quot;, &quot;magrittr&quot;, &quot;stringr&quot;, &quot;purrr&quot;, 
            &quot;here&quot;, &quot;workflowr&quot;, &quot;knitr&quot;, &quot;rmarkdown&quot;, &quot;dabestr&quot;, 
            &quot;RColorBrewer&quot;, &quot;Azimuth&quot;, &quot;DESeq2&quot;, &quot;ggplot2&quot;, 
            &quot;viridis&quot;, &quot;jsonlite&quot;, &quot;glue&quot;),
          file = here::here(&quot;output&quot;, DOCNAME, &quot;packages.bib&quot;))</code></pre>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This methods section details the analytical approaches used in our
study of astrocyte modulation of neuronal development through S100A6
signaling. We employed state-of-the-art single-cell RNA sequencing
analysis techniques to explore gene expression patterns in the
developing mouse cortex, and used robust statistical methods to analyze
MTT assay data measuring astrocyte viability under various conditions.
Our approach emphasizes reproducibility, statistical rigor, and
comprehensive data visualization.</p>
<pre class="r"><code>versions &lt;- list(
    Seurat = packageVersion(&quot;Seurat&quot;),
    dabestr = packageVersion(&quot;dabestr&quot;),
    tidyverse = packageVersion(&quot;tidyverse&quot;),
    RColorBrewer = packageVersion(&quot;RColorBrewer&quot;),
    scCustomize = packageVersion(&quot;scCustomize&quot;),
    SeuratData = packageVersion(&quot;SeuratData&quot;),
    SeuratWrappers = packageVersion(&quot;SeuratWrappers&quot;),
    Azimuth = packageVersion(&quot;Azimuth&quot;),
    cowplot = packageVersion(&quot;cowplot&quot;),
    patchwork = packageVersion(&quot;patchwork&quot;),
    R = R.version.string
)</code></pre>
</div>
<div id="single-cell-rna-sequencing-data-analysis"
class="section level1">
<h1>Single-cell RNA sequencing data analysis</h1>
<p>We analyzed single-cell RNA sequencing data from developing mouse
cortex spanning embryonic day (E) 10 to postnatal day (P) 4. The dataset
was obtained from <span class="citation">(Di Bella et al. 2021)</span>
and accessed through the Single Cell Portal (SCP1290; <span
class="citation">(Tarhan et al., n.d.)</span>). Raw count data and
metadata were downloaded and processed using Seurat (v5.1.0), <span
class="citation">(Satija et al. 2015; Stuart and Satija 2019)</span> in
R (vR version 4.4.0 (2024-04-24)). We chose Seurat for its comprehensive
toolset for quality control, analysis, and exploration of single-cell
RNA-seq data, as well as its wide adoption in the field.</p>
<div id="data-preprocessing-and-quality-control" class="section level2">
<h2>Data preprocessing and quality control</h2>
<p>The raw count matrix was loaded using the <code>Read10X()</code>
function from Seurat. We performed the following preprocessing
steps:</p>
<p>The log1p normalized matrix was converted back to raw counts by
applying <code>expm1()</code>. Scaling factors were calculated based on
the total UMI counts per cell. The count matrix was scaled by
multiplying each cell’s counts by its scaling factor. A new Seurat
object was created using the scaled count matrix. Cells annotated as
doublets, low quality, or red blood cells were removed using the
<code>subset()</code> function. The data was then normalized using the
<code>NormalizeData()</code> function, and 5000 highly variable features
were identified using <code>FindVariableFeatures()</code>.</p>
</div>
<div id="dimensionality-reduction-and-clustering"
class="section level2">
<h2>Dimensionality reduction and clustering</h2>
<p>We performed principal component analysis (PCA) on the variable
features using <code>RunPCA()</code>. Based on the Elbow plot, which
indicates the explained variability of each principal component, we
selected the first 30 out of 50 PCs for downstream analysis. This choice
helps reduce noise in the data while ensuring biological reproducibility
of results.</p>
<p>Uniform Manifold Approximation and Projection (UMAP; <span
class="citation">(McInnes et al. 2018)</span>) and t-distributed
Stochastic Neighbor Embedding (t-SNE; <span class="citation">(Maaten and
Hinton 2008; Kobak and Linderman 2021)</span>) were used for
dimensionality reduction, with embeddings stored in the Seurat object.
Both techniques used the selected 30 PCs as input.</p>
<p>Cells were clustered using the <code>FindNeighbors()</code> and
<code>FindClusters()</code> functions. For community detection, we
employed the Leiden algorithm (resolution = 0.7) instead of the commonly
used Louvain algorithm or alternatives such as walktrap, multilevel, or
infomap. The Leiden algorithm was chosen for its ability to find
converged optimal solutions more efficiently, which is particularly
beneficial for large-scale single-cell datasets <span
class="citation">(Traag, Waltman, and Eck 2019)</span>.</p>
</div>
<div id="gene-expression-analysis" class="section level2">
<h2>Gene expression analysis</h2>
<p>We analyzed the expression of S100 family genes and a curated list of
genes of interest across different developmental stages and cell types.
Feature plots, violin plots, and dot plots were generated using Seurat’s
visualization functions (<code>FeaturePlot()</code>,
<code>VlnPlot()</code>, <code>DotPlot()</code>) and custom functions
from the scCustomize package (v2.1.2).</p>
</div>
<div id="analysis-of-astrocyte-lineage" class="section level2">
<h2>Analysis of astrocyte lineage</h2>
<p>Cells annotated as astrocytes, apical progenitors, and cycling glial
cells were subset for focused analysis of the astrocyte lineage. This
subset was re-clustered using the same approach as described above. We
performed differential expression analysis between astrocyte clusters
using both the <code>FindAllMarkers</code> function in Seurat (using a
logistic regression test; <span class="citation">(Ntranos et al.
2019)</span>) and DESeq2 (v1.44.0), <span class="citation">Love, Huber,
and Anders (2014)</span> on pseudobulk data aggregated by cluster and
developmental stage. The combination of these two approaches allows us
to leverage the strengths of both single-cell and bulk RNA-seq
differential expression methods.</p>
</div>
<div id="visualization" class="section level2">
<h2>Visualization</h2>
<p>Two-dimensional UMAP plots were generated using
<code>FeaturePlot()</code> with the <code>blend = TRUE</code> option to
examine co-expression patterns of key genes. We used custom color
palettes and the patchwork package to create composite figures.</p>
</div>
</div>
<div id="mtt-assay-analysis" class="section level1">
<h1>MTT Assay Analysis</h1>
<p>MTT assay data measuring astrocyte viability after treatment with
eicosapentaenoic acid (EPA) at 5 μM, 10 μM, and 30 μM concentrations was
analyzed using the DABEST (Data Analysis using Bootstrap-Coupled
ESTimation) package v2023.9.12 in R. The analysis was performed to
calculate effect sizes and their confidence intervals using estimation
statistics <span class="citation">(Ho et al. 2019)</span>.</p>
<p>The analysis workflow was as follows:</p>
<ol style="list-style-type: decimal">
<li><p>Data was loaded from a TSV file using <code>read_tsv()</code> and
reshaped into long format using <code>tidyr::gather()</code>.</p></li>
<li><p>For each EPA concentration (5 μM, 10 μM, and 30 μM), control and
treatment groups were compared using the <code>load()</code> function
from DABEST. The data was loaded with the <code>minimeta = TRUE</code>
argument to enable mini-meta analysis across multiple experimental
replicates.</p></li>
<li><p>Mean differences between EPA-treated and control samples were
calculated using the <code>mean_diff()</code> function. This function
computes:</p>
<ul>
<li>The individual mean differences for each experimental replicate</li>
<li>A weighted average of the mean differences (mini-meta delta) using
the generic inverse-variance method</li>
</ul></li>
<li><p>5000 bootstrap resamples were used to generate effect size
estimates with 95% confidence intervals. The confidence intervals are
bias-corrected and accelerated.</p></li>
<li><p>Results were visualized using the <code>dabest_plot()</code>
function to create Cumming estimation plots. These plots show:</p>
<ul>
<li>Raw data points for each group</li>
<li>Group means with 95% confidence intervals</li>
<li>The mean difference for each replicate with its 95% confidence
interval</li>
<li>The weighted mini-meta delta with its 95% confidence interval</li>
</ul></li>
<li><p>Additional statistical information, including p-values from
permutation t-tests, was also calculated and reported, although the
focus of the analysis was on effect sizes and their confidence intervals
rather than null hypothesis significance testing.</p></li>
</ol>
<p>This approach allows for a comprehensive view of the treatment
effects across multiple replicates, taking into account both the
magnitude of the effects and the uncertainty in their estimation. The
mini-meta analysis provides a summary measure of the overall treatment
effect while still preserving information about individual
replicates.</p>
</div>
<div id="other-packages" class="section level1">
<h1>Other packages</h1>
<p>Visualisations and figures were primarily created using the ggplot2
(v), cowplot (v1.1.3) <span class="citation">(Wilke 2024)</span> and
patchwork (v1.2.0.9000) packages using the viridis colour palette (v)
for continuous data. UpSet plots <span class="citation">(Conway, Lex,
and Gehlenborg 2017)</span> were produced using the UpSetR package (v)
<span class="citation">(Gehlenborg 2019)</span> with help from the
gridExtra package (v) <span class="citation">(Auguie 2017)</span>.</p>
<p>Data manipulation was performed using other packages in the tidyverse
(v2.0.0.9000) <span class="citation">(Wickham 2024)</span> particularly
dplyr (v) <span class="citation">(Wickham et al. 2023)</span>, tidyr (v)
<span class="citation">(Wickham, Vaughan, and Girlich 2024)</span> and
purrr (v) <span class="citation">(Wickham and Henry 2023)</span>.</p>
<p>The analysis project was managed using the workflowr (v) <span
class="citation">(Blischak, Carbonetto, and Stephens 2023)</span>
package which was also used to produce the publicly available website
displaying the analysis code, results and output. Reproducible reports
were produced using knitr (v) <span class="citation">(Xie 2024)</span>
and R Markdown (v) <span class="citation">(Allaire et al.
2024)</span>.</p>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>Our methodological approach combines cutting-edge single-cell RNA
sequencing analysis techniques with robust statistical methods for
analyzing experimental data. By using tools like 1) Seurat for scRNA-seq
analysis with two different frameworks for differential gene expression
analysis: logit tailored for the analysis of scRNA-seq data, and DESeq2
on pseudo-bulk data, and 2) DABEST for MTT assay analysis, we ensure a
comprehensive and statistically sound exploration of astrocyte-mediated
neuronal development. The use of estimation statistics and mini-meta
analysis allows for a nuanced interpretation of experimental results,
while our focus on reproducibility and open science practices ensures
that our findings can be thoroughly validated and built upon by the
scientific community.</p>
</div>
<div id="summary" class="section level1">
<h1>Summary</h1>
<p>All analyses were performed using R vR version 4.4.0 (2024-04-24).
Key packages used include Seurat v5.1.0, patchwork v1.2.0.9000, ggplot2
v3.5.1, dplyr v1.1.4, and DABEST v2023.9.12. Code for the full analysis
is available at <a href="https://github.com/harkany-lab/Cinquina_2024"
class="uri">https://github.com/harkany-lab/Cinquina_2024</a>.</p>
<div id="output-files" class="section level2">
<h2>Output files</h2>
<pre class="r"><code>versions &lt;- purrr::map(versions, as.character)
versions &lt;- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here(&quot;output&quot;, DOCNAME, &quot;package-versions.json&quot;))</code></pre>
</div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent"
entry-spacing="0">
<div id="ref-R-rmarkdown" class="csl-entry">
Allaire, JJ, Yihui Xie, Christophe Dervieux, Jonathan McPherson, Javier
Luraschi, Kevin Ushey, Aron Atkins, et al. 2024. <em>Rmarkdown: Dynamic
Documents for r</em>. <a
href="https://github.com/rstudio/rmarkdown">https://github.com/rstudio/rmarkdown</a>.
</div>
<div id="ref-R-gridExtra" class="csl-entry">
Auguie, Baptiste. 2017. <em>gridExtra: Miscellaneous Functions for
"Grid" Graphics</em>.
</div>
<div id="ref-R-workflowr" class="csl-entry">
Blischak, John, Peter Carbonetto, and Matthew Stephens. 2023.
<em>Workflowr: A Framework for Reproducible and Collaborative Data
Science</em>. <a
href="https://github.com/workflowr/workflowr">https://github.com/workflowr/workflowr</a>.
</div>
<div id="ref-conwayUpSetRPackageVisualization2017" class="csl-entry">
Conway, Jake R, Alexander Lex, and Nils Gehlenborg. 2017. <span>“UpSetR:
An R Package for the Visualization of Intersecting Sets and Their
Properties.”</span> Edited by John Hancock. <em>Bioinformatics</em> 33
(18): 2938–40. <a
href="https://doi.org/10.1093/bioinformatics/btx364">https://doi.org/10.1093/bioinformatics/btx364</a>.
</div>
<div id="ref-dibellaMolecularLogicCellular2021" class="csl-entry">
Di Bella, Daniela J., Ehsan Habibi, Robert R. Stickels, Gabriele Scalia,
Juliana Brown, Payman Yadollahpour, Sung Min Yang, et al. 2021.
<span>“Molecular logic of cellular diversification in the mouse cerebral
cortex.”</span> <em>Nature</em> 595 (7868): 554–59. <a
href="https://doi.org/10.1038/s41586-021-03670-5">https://doi.org/10.1038/s41586-021-03670-5</a>.
</div>
<div id="ref-R-UpSetR" class="csl-entry">
Gehlenborg, Nils. 2019. <em>UpSetR: A More Scalable Alternative to Venn
and Euler Diagrams for Visualizing Intersecting Sets</em>. <a
href="http://github.com/hms-dbmi/UpSetR">http://github.com/hms-dbmi/UpSetR</a>.
</div>
<div id="ref-hoMovingValuesData2019" class="csl-entry">
Ho, Joses, Tayfun Tumkaya, Sameer Aryal, Hyungwon Choi, and Adam
Claridge-Chang. 2019. <span>“Moving beyond P values: data analysis with
estimation graphics.”</span> <em>Nature Methods</em> 16 (7): 565–66. <a
href="https://doi.org/10.1038/s41592-019-0470-3">https://doi.org/10.1038/s41592-019-0470-3</a>.
</div>
<div id="ref-kobakInitializationCriticalPreserving2021"
class="csl-entry">
Kobak, Dmitry, and George C. Linderman. 2021. <span>“Initialization Is
Critical for Preserving Global Data Structure in Both t-SNE and
UMAP.”</span> <em>Nature Biotechnology</em> 39 (2): 156–57. <a
href="https://doi.org/ghxbbn">https://doi.org/ghxbbn</a>.
</div>
<div id="ref-loveModeratedEstimationFold2014" class="csl-entry">
Love, Michael I., Wolfgang Huber, and Simon Anders. 2014.
<span>“Moderated estimation of fold change and dispersion for RNA-seq
data with DESeq2.”</span> <em>Genome Biology</em> 15 (12): 550. <a
href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-maatenVisualizingDataUsing2008" class="csl-entry">
Maaten, Laurens van der, and Geoffrey Hinton. 2008. <span>“Visualizing
Data Using t-SNE.”</span> <em>Journal of Machine Learning Research</em>
9 (86): 2579–2605. <a
href="http://jmlr.org/papers/v9/vandermaaten08a.html">http://jmlr.org/papers/v9/vandermaaten08a.html</a>.
</div>
<div id="ref-mcinnesUMAPUniformManifold2018" class="csl-entry">
McInnes, Leland, John Healy, Nathaniel Saul, and Lukas Großberger. 2018.
<span>“UMAP: Uniform Manifold Approximation and Projection.”</span>
<em>Journal of Open Source Software</em> 3 (29): 861. <a
href="https://doi.org/10.21105/joss.00861">https://doi.org/10.21105/joss.00861</a>.
</div>
<div id="ref-ntranosDiscriminativeLearningApproach2019"
class="csl-entry">
Ntranos, Vasilis, Lynn Yi, Páll Melsted, and Lior Pachter. 2019.
<span>“A Discriminative Learning Approach to Differential Expression
Analysis for Single-Cell RNA-Seq.”</span> <em>Nature Methods</em> 16
(2): 163–66. <a
href="https://doi.org/gftsnv">https://doi.org/gftsnv</a>.
</div>
<div id="ref-satijaSpatialReconstructionSinglecell2015"
class="csl-entry">
Satija, Rahul, Jeffrey A. Farrell, David Gennert, Alexander F. Schier,
and Aviv Regev. 2015. <span>“Spatial Reconstruction of Single-Cell Gene
Expression Data.”</span> <em>Nature Biotechnology</em> 33 (5): 495–502.
<a
href="https://doi.org/10.1038/nbt.3192">https://doi.org/10.1038/nbt.3192</a>.
</div>
<div id="ref-stuartIntegrativeSinglecellAnalysis2019" class="csl-entry">
Stuart, Tim, and Rahul Satija. 2019. <span>“Integrative Single-Cell
Analysis.”</span> <em>Nature Reviews. Genetics</em> 20 (5): 257–72. <a
href="https://doi.org/10.1038/s41576-019-0093-7">https://doi.org/10.1038/s41576-019-0093-7</a>.
</div>
<div id="ref-tarhanSingleCellPortal2023" class="csl-entry">
Tarhan, Leyla, Jon Bistline, Jean Chang, Bryan Galloway, Emily Hanna,
and Eric Weitz. n.d. <span>“Single Cell Portal: An Interactive Home for
Single-Cell Genomics Data.”</span> <a
href="https://doi.org/10.1101/2023.07.13.548886">https://doi.org/10.1101/2023.07.13.548886</a>.
</div>
<div id="ref-traagLouvainLeidenGuaranteeing2019" class="csl-entry">
Traag, V. A., L. Waltman, and N. J. van Eck. 2019. <span>“From Louvain
to Leiden: Guaranteeing Well-Connected Communities.”</span>
<em>Scientific Reports</em> 9 (1): 5233. <a
href="https://doi.org/10.1038/s41598-019-41695-z">https://doi.org/10.1038/s41598-019-41695-z</a>.
</div>
<div id="ref-R-tidyverse" class="csl-entry">
Wickham, Hadley. 2024. <em>Tidyverse: Easily Install and Load the
Tidyverse</em>. <a
href="https://tidyverse.tidyverse.org">https://tidyverse.tidyverse.org</a>.
</div>
<div id="ref-R-dplyr" class="csl-entry">
Wickham, Hadley, Romain François, Lionel Henry, Kirill Müller, and Davis
Vaughan. 2023. <em>Dplyr: A Grammar of Data Manipulation</em>. <a
href="https://dplyr.tidyverse.org">https://dplyr.tidyverse.org</a>.
</div>
<div id="ref-R-purrr" class="csl-entry">
Wickham, Hadley, and Lionel Henry. 2023. <em>Purrr: Functional
Programming Tools</em>. <a
href="https://purrr.tidyverse.org/">https://purrr.tidyverse.org/</a>.
</div>
<div id="ref-R-tidyr" class="csl-entry">
Wickham, Hadley, Davis Vaughan, and Maximilian Girlich. 2024. <em>Tidyr:
Tidy Messy Data</em>. <a
href="https://tidyr.tidyverse.org">https://tidyr.tidyverse.org</a>.
</div>
<div id="ref-R-cowplot" class="csl-entry">
Wilke, Claus O. 2024. <em>Cowplot: Streamlined Plot Theme and Plot
Annotations for Ggplot2</em>. <a
href="https://wilkelab.org/cowplot/">https://wilkelab.org/cowplot/</a>.
</div>
<div id="ref-R-knitr" class="csl-entry">
Xie, Yihui. 2024. <em>Knitr: A General-Purpose Package for Dynamic
Report Generation in r</em>. <a
href="https://yihui.org/knitr/">https://yihui.org/knitr/</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] patchwork_1.2.0.9000  cowplot_1.1.3         magrittr_2.0.3       
 [4] Azimuth_0.5.0         shinyBS_0.61.1        SeuratWrappers_0.3.5 
 [7] SeuratData_0.2.2.9001 scCustomize_2.1.2     RColorBrewer_1.1-3   
[10] here_1.0.1            dabestr_2023.9.12     Seurat_5.1.0         
[13] SeuratObject_5.0.2    sp_2.1-4              lubridate_1.9.3      
[16] forcats_1.0.0         stringr_1.5.1         dplyr_1.1.4          
[19] purrr_1.0.2           readr_2.1.5           tidyr_1.3.1          
[22] tibble_3.2.1          ggplot2_3.5.1         tidyverse_2.0.0.9000 
[25] jsonlite_1.8.8        knitr_1.47            glue_1.7.0           
[28] workflowr_1.7.1      

loaded via a namespace (and not attached):
  [1] IRanges_2.38.0                    R.methodsS3_1.8.2                
  [3] vroom_1.6.5                       poweRlaw_0.80.0                  
  [5] goftest_1.2-3                     DT_0.33                          
  [7] Biostrings_2.72.1                 vctrs_0.6.5                      
  [9] spatstat.random_3.2-3             digest_0.6.35                    
 [11] png_0.1-8                         shape_1.4.6.1                    
 [13] git2r_0.33.0                      ggrepel_0.9.5.9999               
 [15] deldir_2.0-4                      parallelly_1.37.1                
 [17] MASS_7.3-61                       Signac_1.13.9003                 
 [19] reshape2_1.4.4                    httpuv_1.6.15                    
 [21] BiocGenerics_0.50.0               withr_3.0.0                      
 [23] ggrastr_1.0.2                     xfun_0.45                        
 [25] survival_3.7-0                    EnsDb.Hsapiens.v86_2.99.0        
 [27] memoise_2.0.1                     ggbeeswarm_0.7.2                 
 [29] janitor_2.2.0.9000                zoo_1.8-12                       
 [31] GlobalOptions_0.1.2               gtools_3.9.5                     
 [33] pbapply_1.7-2                     R.oo_1.26.0                      
 [35] rematch2_2.1.2                    KEGGREST_1.44.0                  
 [37] promises_1.3.0                    httr_1.4.7                       
 [39] restfulr_0.0.15                   globals_0.16.3                   
 [41] fitdistrplus_1.1-11               rhdf5filters_1.16.0              
 [43] ps_1.7.6                          rhdf5_2.48.0                     
 [45] rstudioapi_0.16.0                 UCSC.utils_1.0.0                 
 [47] miniUI_0.1.1.1                    generics_0.1.3                   
 [49] processx_3.8.4                    curl_5.2.1                       
 [51] S4Vectors_0.42.0                  zlibbioc_1.50.0                  
 [53] polyclip_1.10-6                   GenomeInfoDbData_1.2.12          
 [55] SparseArray_1.4.8                 xtable_1.8-4                     
 [57] pracma_2.4.4                      evaluate_0.24.0                  
 [59] S4Arrays_1.4.1                    hms_1.1.3                        
 [61] GenomicRanges_1.56.1              irlba_2.3.5.1                    
 [63] colorspace_2.1-0                  hdf5r_1.3.10                     
 [65] ROCR_1.0-11                       reticulate_1.37.0                
 [67] spatstat.data_3.0-4               lmtest_0.9-40                    
 [69] snakecase_0.11.1                  later_1.3.2                      
 [71] lattice_0.22-6                    spatstat.geom_3.2-9              
 [73] future.apply_1.11.2               getPass_0.2-4                    
 [75] scattermore_1.2                   XML_3.99-0.16.1                  
 [77] matrixStats_1.3.0                 RcppAnnoy_0.0.22                 
 [79] pillar_1.9.0                      nlme_3.1-165                     
 [81] pwalign_1.0.0                     caTools_1.18.2                   
 [83] compiler_4.4.0                    RSpectra_0.16-1                  
 [85] stringi_1.8.4                     tensor_1.5                       
 [87] SummarizedExperiment_1.34.0       GenomicAlignments_1.40.0         
 [89] plyr_1.8.9                        crayon_1.5.2                     
 [91] abind_1.4-5                       BiocIO_1.14.0                    
 [93] googledrive_2.1.1                 bit_4.0.5                        
 [95] fastmatch_1.1-4                   whisker_0.4.1                    
 [97] codetools_0.2-20                  bslib_0.7.0                      
 [99] paletteer_1.6.0                   plotly_4.10.4                    
[101] mime_0.12                         splines_4.4.0                    
[103] circlize_0.4.16                   Rcpp_1.0.12                      
[105] fastDummies_1.7.3                 cellranger_1.1.0                 
[107] blob_1.2.4                        utf8_1.2.4                       
[109] seqLogo_1.70.0                    AnnotationFilter_1.28.0          
[111] fs_1.6.4                          listenv_0.9.1                    
[113] Matrix_1.7-0                      callr_3.7.6                      
[115] tzdb_0.4.0                        pkgconfig_2.0.3                  
[117] tools_4.4.0                       cachem_1.1.0                     
[119] RSQLite_2.3.7                     viridisLite_0.4.2                
[121] DBI_1.2.3                         fastmap_1.2.0                    
[123] rmarkdown_2.27                    scales_1.3.0                     
[125] grid_4.4.0                        ica_1.0-3                        
[127] shinydashboard_0.7.2              Rsamtools_2.20.0                 
[129] sass_0.4.9                        ggprism_1.0.5                    
[131] BiocManager_1.30.23               dotCall64_1.1-1                  
[133] RANN_2.6.1                        yaml_2.3.8                       
[135] MatrixGenerics_1.16.0             rtracklayer_1.64.0               
[137] cli_3.6.2                         stats4_4.4.0                     
[139] leiden_0.4.3.1                    lifecycle_1.0.4                  
[141] uwot_0.2.2                        Biobase_2.64.0                   
[143] presto_1.0.0                      BSgenome.Hsapiens.UCSC.hg38_1.4.5
[145] BiocParallel_1.38.0               annotate_1.82.0                  
[147] timechange_0.3.0                  gtable_0.3.5                     
[149] rjson_0.2.21                      ggridges_0.5.6                   
[151] progressr_0.14.0                  parallel_4.4.0                   
[153] RcppHNSW_0.6.0                    TFBSTools_1.42.0                 
[155] bitops_1.0-7                      bit64_4.0.5                      
[157] Rtsne_0.17                        spatstat.utils_3.0-5             
[159] CNEr_1.40.0                       jquerylib_0.1.4                  
[161] shinyjs_2.1.0                     SeuratDisk_0.0.0.9021            
[163] R.utils_2.12.3                    lazyeval_0.2.2                   
[165] shiny_1.8.1.1                     htmltools_0.5.8.1                
[167] GO.db_3.19.1                      sctransform_0.4.1                
[169] rappdirs_0.3.3                    ensembldb_2.28.0                 
[171] TFMPvalue_0.0.9                   spam_2.10-0                      
[173] googlesheets4_1.1.1               XVector_0.44.0                   
[175] RCurl_1.98-1.14                   rprojroot_2.0.4                  
[177] BSgenome_1.72.0                   gridExtra_2.3                    
[179] JASPAR2020_0.99.10                igraph_2.0.3                     
[181] R6_2.5.1                          RcppRoll_0.3.0                   
[183] GenomicFeatures_1.56.0            cluster_2.1.6                    
[185] Rhdf5lib_1.26.0                   gargle_1.5.2                     
[187] GenomeInfoDb_1.40.1               DirichletMultinomial_1.46.0      
[189] DelayedArray_0.30.1               tidyselect_1.2.1                 
[191] vipor_0.4.7                       ProtGenerics_1.36.0              
[193] AnnotationDbi_1.66.0              future_1.33.2                    
[195] rsvd_1.0.5                        munsell_0.5.1                    
[197] KernSmooth_2.23-24                data.table_1.15.4                
[199] htmlwidgets_1.6.4                 rlang_1.1.4                      
[201] spatstat.sparse_3.0-3             spatstat.explore_3.2-7           
[203] remotes_2.5.0                     fansi_1.0.6                      
[205] beeswarm_0.4.0                   </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
